tests/random_test.py

```diff
@@ -1032,6 +1032,27 @@ def testLogistic(self, dtype):
     for samples in [uncompiled_samples, compiled_samples]:
       self._CheckKolmogorovSmirnovCDF(samples, scipy.stats.logistic().cdf)
 
+  @parameterized.named_parameters(jtu.cases_from_list(
+      {"testcase_name": "_n={}_shape={}"\
+       .format(n, jtu.format_shape_dtype_string(shape, dtype)),
+       "n": n,
+       "shape": shape,
+       "dtype": dtype}
+      for n in range(1, 5)
+      for shape in [(), (5,), (10, 5)]
+      for dtype in jtu.dtypes.floating + jtu.dtypes.complex))
+  def testOrthogonal(self, n, shape, dtype):
+    key = self.seed_prng(0)
+    q = random.orthogonal(key, n, shape, dtype)
+    self.assertEqual(q.shape, (*shape, n, n))
+    self.assertEqual(q.dtype, dtype)
+    with jax.numpy_rank_promotion('allow'):
+      self.assertAllClose(
+        jnp.einsum('...ij,...jk->...ik', q, jnp.conj(q).swapaxes(-2, -1)),
+        jnp.broadcast_to(jnp.eye(n, dtype=dtype), (*shape, n, n)),
+        atol=1e-6,
```

Comment:
This test is failing on TPU backends; I suspect it might be because you set `atol` to a constant, which overrides the default behavior of adjusting the tolerance to fit the particular backend and dtype. Can we try removing this?

We may need to adjust the tolerance further for TPU backends, I'm not sure.